omod/src/main/java/org/openmrs/module/owa/web/controller/OwaRestController.java

```diff
@@ -163,7 +163,7 @@
 				if (!url.isEmpty()) {
 					InputStream inputStream = ModuleUtil.getURLStream(downloadUrl);
 					log.warn("url pathname: " + downloadUrl.getPath());
-					String fileName = downloadUrl.getQuery().substring(downloadUrl.getQuery().lastIndexOf("=") + 1);
+					String fileName = urlObject.getFileName() != null ? urlObject.getFileName() + ".zip" : downloadUrl.getQuery().substring(downloadUrl.getQuery().lastIndexOf("=") + 1);
 					File file = ModuleUtil.insertModuleFile(inputStream, fileName);
```

Comment:
I think you should refactor `String fileName = urlObject.getFileName() != null` to `String fileName = !urlObject.getFileName()`

Reply:
@Charpell So even though a `String` typed variable can automatically coerce to a `Boolean` in javascript, the same principle does not necessarily apply to java. Why I did not automatically use the `Not` operator on the string.